scripts/commit_validation/commit_validation/validators/crc_validator.py

```diff
@@ -0,0 +1,48 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+import binascii
+import fnmatch
+import os.path
+import re
+from typing import Type, List
+
+from commit_validation.commit_validation import Commit, CommitValidator, SOURCE_FILE_EXTENSIONS, EXCLUDED_VALIDATION_PATTERNS, VERBOSE
+
+class CrcValidator(CommitValidator):
+    """A file-level validator that makes sure a file does not contain an invalid CRC"""
+
+    def run(self, commit: Commit, errors: List[str]) -> bool:
+        for file_name in commit.get_files():
+            for pattern in EXCLUDED_VALIDATION_PATTERNS:
+                if fnmatch.fnmatch(file_name, pattern):
+                    if VERBOSE: print(f'{file_name}::{self.__class__.__name__} SKIPPED - Validation pattern excluded on path.')
+                    break
+            else:
+                if os.path.splitext(file_name)[1].lower() not in SOURCE_FILE_EXTENSIONS:
+                    if VERBOSE: print(f'{file_name}::{self.__class__.__name__} SKIPPED - File excluded based on extension.')
+                    continue
+                
+                with open(file_name, mode='r', encoding='utf8') as fh:
+                    fileContents = fh.read()
+                    matchesFound = re.findall(r'AZ_CRC\("([^"]+)",([^\)]*)\)', fileContents)
+                    for element in matchesFound:
+                        stringInCode = element[0]
+                        valueInCode = element[1].strip()
+                        expectedValue = "{0:#0{1}x}".format(binascii.crc32(stringInCode.lower().encode('utf8')), 10)
```

Comment:
You can use `hex(binascii.crc32(<crc32>) & 0xffffffff)` t oconvert the value to a string as well

Reply:
actually, this trims trailing `0` from the string

Reply:
You are correct.

But that provided me with another thought
Shouldn't we be comparing the actual numeric values?

Reply:
I think we also want to keep it easier to searches, and that is how its being printed. If someone where to put a decimal value there, it would make the search harder.

Reply:
For correctness however a decimal value that is technically correct would cause the validator to fail, due it not looking exactly like the string of "0x<8-hex-digits>"

This check then become very specific to a format of a string and not the actual value.

For example if the CRC isn't isn't 8-hex-digits, then the validator would fail, despite the values being correct.

Plus since we don't have a tool to generate the CRC32 values anymore, then we can't assume that the format the user enters in for them will be of the format "0x<8-hex-digits>"